### PR TITLE
feat: Server.WithEngine config

### PR DESCRIPTION
**Pull request by dylanhitt**
Opened a PR to give a look at the WithEngine concept. 

I personally like this bit better than the WithEngineOptions as we're able to separate the the two different types of `OpenAPIConfig` and `OpenAPIServerConfig`. It also will make for less impactful if we decide to move more `OpenAPIServerConfig` options into the engine `OpenAPIConfig`.

Also having such a well tested code base makes this code so nice to work with

### CODE REVIEW

File: server.go [hunk 1]
```diff
@@ -365,14 +363,36 @@ func WithoutLogger() func(*Server) {
...
+
+		s.OpenAPIServerConfig.DisableSwaggerUI = config.DisableSwaggerUI
+
+		if !validateJsonSpecUrl(s.OpenAPIServerConfig.JsonURL) {
+			slog.Error("Error serving openapi json spec. Value of 's.OpenAPIServerConfig.JsonSpecUrl' option is not valid", "url", s.OpenAPIServerConfig.JsonURL)
+			return
+		}
+
+		if !validateSwaggerUrl(s.OpenAPIServerConfig.SwaggerURL) {
+			slog.Error("Error serving swagger ui. Value of 's.OpenAPIServerConfig.SwaggerUrl' option is not valid", "url", s.OpenAPIServerConfig.SwaggerURL)
```
Comment (dylanhitt) [inline server.go:386]:
```suggestion
			slog.Error("Error serving swagger ui. Value of 's.OpenAPIServerConfig.SwaggerURL' option is not valid", "url", s.OpenAPIServerConfig.SwaggerURL)
```

File: server.go [hunk 2]
```diff
@@ -365,14 +363,36 @@ func WithoutLogger() func(*Server) {
...
+			s.OpenAPIServerConfig.SwaggerURL = config.SwaggerURL
+		}
+		if config.UIHandler != nil {
+			s.OpenAPIServerConfig.UIHandler = config.UIHandler
+		}
+
+		s.OpenAPIServerConfig.DisableSwaggerUI = config.DisableSwaggerUI
+
+		if !validateJsonSpecUrl(s.OpenAPIServerConfig.JsonURL) {
+			slog.Error("Error serving openapi json spec. Value of 's.OpenAPIServerConfig.JsonSpecUrl' option is not valid", "url", s.OpenAPIServerConfig.JsonURL)
```
Comment (dylanhitt) [inline server.go:381]:
```suggestion
			slog.Error("Error serving openapi json spec. Value of 's.OpenAPIServerConfig.JsonSpecURL' option is not valid", "url", s.OpenAPIServerConfig.JsonURL)
```

Comment (dylanhitt) [inline server.go:381]:
```suggestion
			slog.Error("Error serving openapi json spec. Value of 's.OpenAPIServerConfig.JsonSpecURL' option is not valid", "url", s.OpenAPIServerConfig.JsonURL)
```

File: openapi_test.go [hunk 3]
```diff
@@ -270,16 +268,14 @@ func TestServer_OutputOpenApiSpec(t *testing.T) {
 	})
 	t.Run("do not print file", func(t *testing.T) {
 		s := NewServer(
-			WithEngineOptions(
+			WithEngine(NewEngine(
```
Comment (ccoVeille) [inline openapi_test.go:271]: What about a helper like this

```suggestion
			WithNewEngine(
```

Do you really need a new engine BTW?

Comment (dylanhitt) [reply openapi_test.go:271]: Yes that’s better

Comment (dylanhitt) [reply openapi_test.go:271]: And what do you mean do I need a “NewEngine”. The whole purpose of this PR is to demonstrate the difference between WithEngineOptions and WithEngine. Do you have another pattern in mind that you would suggest?

Comment (dylanhitt) [reply openapi_test.go:271]: I guess looking at it. WithEngine is more redundant than the WithEngineOptions. 

I do like the splitting of the config that pertinent to the Engine and pertinent to the net/http server. I think imma update the other to split those out.

**Comment by dylanhitt** [comment]
Closing favor of #302

File: openapi.go [hunk 4]
```diff
@@ -116,24 +116,24 @@ func (s *Server) OutputOpenAPISpec() openapi3.T {
...
-	GetStd(s, s.OpenAPIConfig.JsonURL, func(w http.ResponseWriter, r *http.Request) {
+	GetStd(s, s.OpenAPIServerConfig.JsonURL, func(w http.ResponseWriter, r *http.Request) {
 		w.Header().Set("Content-Type", "application/json")
 		_, _ = w.Write(jsonSpec)
 	})
-	s.printOpenAPIMessage(fmt.Sprintf("JSON spec: %s%s", s.url(), s.OpenAPIConfig.JsonURL))
+	s.printOpenAPIMessage(fmt.Sprintf("JSON spec: %s%s", s.url(), s.OpenAPIServerConfig.JsonURL))
 
-	if !s.OpenAPIConfig.DisableSwaggerUI {
+	if !s.OpenAPIServerConfig.DisableSwaggerUI {
```
[the remark this answers is not in the document]
Comment (dylanhitt) [reply openapi.go:125]: Implemented in #302

**Pull request closed without merging**